### PR TITLE
State what happens when metric names clash

### PR DESCRIPTION
**Pull request by Azquelt**
I'm not sure how specific we should be here.

We could go all the way and explain how each metric type should be combined - in most cases this is a simple sum. We could then add tests to ensure that metrics are combined correctly in all cases.

From an implementation point of view, this isn't too arduous, though some additional work would be required to combine gauges.

However, I'm not sure how valuable combining metrics is for the user. Essentially we're just trying to do something sensible if this situation happens so for the moment, I've just added a simple test which ensures that the implementation doesn't throw an exception and increments the basic invocation metric.

### CODE REVIEW

File: spec/src/main/asciidoc/metrics.asciidoc [hunk 1]
```diff
@@ -28,6 +28,8 @@ the methods annotated with a `@Retry`, `@Timeout`, `@CircuitBreaker` or `@Bulkhe
 The automatically added metrics follow a consistent pattern which includes the fully qualified name of the annotated method.
 In the tables below, the placeholder `<name>` should be replaced by the fully qualified method name.
 
+If two methods have the same fully qualified name then the metrics for those methods will be combined.
```
Comment (OndrejM) [inline spec/src/main/asciidoc/metrics.asciidoc]: I would at least explain that the end result is not portable and may vary across implementations. For portable behavior, monitored methods in the same class should have unique names.

Comment (Azquelt) [reply spec/src/main/asciidoc/metrics.asciidoc]: Makes sense, I will add that.

Comment (Azquelt) [reply spec/src/main/asciidoc/metrics.asciidoc]: Done.

**Comment by Azquelt** [comment]
I have made the requested changes and rebased on master to resolve a conflict.

**Comment by OndrejM** [comment]
I'm merging this as it's not anything huge,  more or less a clarification

**Pull request merged**